RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ChromeCacheExtractor.java

```diff
@@ -0,0 +1,1265 @@
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardArtifact.ARTIFACT_TYPE;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.DerivedFile;
+import org.sleuthkit.datamodel.ReadContentInputStream;
+import org.sleuthkit.datamodel.TimeUtilities;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskData;
+import org.sleuthkit.datamodel.TskException;
+
+/**
+ * Extracts and parses Chrome Cache files.
+ * 
+ * Cache may hold images, scripts, CSS, JSON files, 
+ * and the URL they were downloaded from.
+ * 
+ * Cache entries may or may not be compressed, 
+ * and the entries may reside in container files or external files.
+ * 
+ * We extract cache entries, create derived files if needed, 
+ * and record the URL.
+ */
+final class ChromeCacheExtractor {
+    
+    private final static String DEFAULT_CACHE_STR = "default/cache";
+    private final static String BROTLI_MIMETYPE ="application/x-brotli";
+    
+    private final static long UINT32_MASK = 0xFFFFFFFFl;
+    
+    private final static int INDEXFILE_HDR_SIZE = 92*4;
+    private final static int DATAFILE_HDR_SIZE = 8192;
+    
+    private final static Logger logger = Logger.getLogger(ChromeCacheExtractor.class.getName());
+    
+    private static final String VERSION_NUMBER = "1.0.0";
+    private final String moduleName;
+    
+    private String absOutputFolderName;
+    private String relOutputFolderName;
+     
+    private final Content dataSource;
+    private final IngestJobContext context;
+    private final IngestServices services = IngestServices.getInstance();
+    private Case currentCase;
+    private FileManager fileManager;
+ 
+    
+    private Map<String, CacheFileCopy> filesTable = new HashMap<>();
+    
+    /**
+     * Encapsulates  abstract file for a cache file as well as a temp file copy
+     * that can be accessed as a random access file.
+     */
+    final class CacheFileCopy {
+        
+        private final AbstractFile abstractFile;
+        private final RandomAccessFile fileCopy;
+        private final ByteBuffer byteBuffer;
+
+        CacheFileCopy (AbstractFile abstractFile, RandomAccessFile fileCopy, ByteBuffer buffer ) {
+            this.abstractFile = abstractFile;
+            this.fileCopy = fileCopy;
+            this.byteBuffer = buffer;
+        }
+        
+        public RandomAccessFile getFileCopy() {
+            return fileCopy;
+        }
+        public ByteBuffer getByteBuffer() {
+            return byteBuffer;
+        }
+        AbstractFile getAbstractFile() {
+            return abstractFile;
+        } 
+    }
+
+    ChromeCacheExtractor(Content dataSource, IngestJobContext context ) {
+        moduleName = NbBundle.getMessage(ChromeCacheExtractor.class, "ChromeCacheExtractor.moduleName");           
```

Comment:
Should use the @message annotation instead of manually modifying Bundles.properties.